dali/base/dafdesc.cpp

```diff
@@ -1340,6 +1352,13 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
             if (_interleaved)
                 *_interleaved = interleaved;
         }
+        const char * buffer = mb.toByteArray()+mb.getPos();
+        const char * historyStr = "History";
+        bool hasEnoughData = (mb.length() - mb.getPos()) > strlen(historyStr);
```

Comment:
I'd be very tempted to add a true version # at this point (where serialized), so that future additions can check it and conditionally deserialize based on version #.